src/m365/spo/commands/commandset/commandset-add.ts

```diff
@@ -0,0 +1,168 @@
+
+  public get name(): string {
+    return commands.COMMANDSET_ADD;
+  }
+
+  public get description(): string {
+    return 'Add a ListView Command Set to a site.';
+  }
+
+  constructor() {
+    super();
+
+    this.#initTelemetry();
+    this.#initOptions();
+    this.#initValidators();
+  }
+
+  #initTelemetry(): void {
+    this.telemetry.push((args: CommandArgs) => {
+      Object.assign(this.telemetryProperties, {
+        clientSideComponentProperties: typeof args.options.clientSideComponentProperties !== 'undefined',
+        scope: typeof args.options.scope !== 'undefined',
+        location: typeof args.options.location !== 'undefined'
+      });
+    });
+  }
+
+  #initOptions(): void {
+    this.options.unshift(
+      {
+        option: '-t, --title <title>'
+      },
+      {
+        option: '-u, --webUrl <webUrl>'
+      },
+      {
+        option: '-l, --listType <listType>', autocomplete: SpoCommandsetAddCommand.listTypes
+      },
+      {
+        option: '-i, --clientSideComponentId  <clientSideComponentId>'
+      },
+      {
+        option: '--clientSideComponentProperties  [clientSideComponentProperties]'
+      },
+      {
+        option: '-s, --scope [scope]', autocomplete: SpoCommandsetAddCommand.scopes
+      },
+      {
+        option: '--location [location]', autocomplete: SpoCommandsetAddCommand.locations
+      }
+    );
+  }
+
+  #initValidators(): void {
+    this.validators.push(
+      async (args: CommandArgs) => {
+        if (args.options.clientSideComponentId && !validation.isValidGuid(args.options.clientSideComponentId as string)) {
+          return `${args.options.clientSideComponentId} is not a valid GUID`;
+        }
+
+        if (SpoCommandsetAddCommand.listTypes.indexOf(args.options.listType) < 0) {
+          return `${args.options.listType} is not a valid list type. Allowed values are ${SpoCommandsetAddCommand.listTypes.join(', ')}`;
+        }
+
+        if (args.options.scope && SpoCommandsetAddCommand.scopes.indexOf(args.options.scope) < 0) {
+          return `${args.options.scope} is not a valid scope. Allowed values are ${SpoCommandsetAddCommand.scopes.join(', ')}`;
+        }
+
+        if (args.options.location && SpoCommandsetAddCommand.locations.indexOf(args.options.location) < 0) {
+          return `${args.options.location} is not a valid location. Allowed values are ${SpoCommandsetAddCommand.locations.join(', ')}`;
+        }
+
+        return validation.isValidSharePointUrl(args.options.webUrl);
+      }
+    );
+  }
+
+  public async commandAction(logger: Logger, args: CommandArgs): Promise<void> {
+    if (this.verbose) {
+      logger.logToStderr(`Adding ListView Command Set ${args.options.clientSideComponentId} to site '${args.options.webUrl}'...`);
+    }
+
+    if (!args.options.scope) {
+      args.options.scope = 'Site';
+    }
+
+    const location: string = this.getLocation(args.options.location ? args.options.location : '');
```

Comment:
I would actually here handle default value not do it as part of the `switch/case`